extensions/wikia/Search/WikiaSearchController.class.php

```diff
@@ -525,20 +525,24 @@ private function useGoSearch() {
 	 * @return \Wikia\Search\Config
 	 */
 	protected function getSearchConfigFromRequest() {
-		$request = $this->getRequest();
+		global $wgRequest;
+
+		// Use WebRequest instead of Nirvana request
+		// Nirvana request does not process certain Unicode stuff correctly which causes HTTP 500
+		$request = $wgRequest;
 		$searchConfig = new Wikia\Search\Config();
 		$resultsPerPage = $this->isCorporateWiki() ? self::INTERWIKI_RESULTS_PER_PAGE : self::RESULTS_PER_PAGE;
 		$resultsPerPage = empty( $this->wg->SearchResultsPerPage ) ? $resultsPerPage : $this->wg->SearchResultsPerPage;
 		$searchConfig
 			->setQuery                   ( $request->getVal( 'query', $request->getVal( 'search' ) ) )
 			->setCityId                  ( $this->wg->CityId )
 			->setLimit                   ( $request->getInt( 'limit', $resultsPerPage ) )
-			->setPage                    ( $request->getVal( 'page', 1) )
+			->setPage                    ( $request->getInt( 'page', 1 ) )
 			->setRank                    ( $request->getVal( 'rank', 'default' ) )
-			->setHub                     ( $request->getVal( 'hub', false ) )
+			->setHub                     ( $request->getBool( 'hub', false ) )
 			->setInterWiki               ( $this->isCorporateWiki() )
-			->setVideoSearch             ( $request->getVal( 'videoSearch', false ) )
-			->setFilterQueriesFromCodes  ( $request->getVal( 'filters', array() ) )
+			->setVideoSearch             ( $request->getBool( 'videoSearch', false ) )
+			->setFilterQueriesFromCodes  ( $request->getArray( 'filters', [] ) )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "filters" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)